### ISSUE TITLE
isValidating stays true after 2nd failed request

### ISSUE THREAD

**Issue by mtakac** [vercel/swr#280]
I am trying to implement login using swr, but if my login request fails 2 times in a row `isValidating` gets stuck and returns `true` even after the request alredy finished.

### Expected behaviour:
I try to login with incorrect credentials
Login request fails
isValidating is set to false no matter how many times the request already failed

### Current behaviour:
I try to login with incorrect credentials
Login request fails
isValidating is set to false
But if I try to login with incorrect credentials again
Login request fails
isValidating is *not set to false* anymore

This is my login component (simplified):
```
const LoginForm = React.FC () => {
  const [email, setEmail] = useState<string>();
  const [password, setPassword] = useState<string>();
  const [shouldSubmit, setShouldSubmit] = useState(false);

  const { data, error, isValidating } = useSWR(
    shouldSubmit ? ['/signIn', email, password] : null,
    (url, email, password) => authenticationApi.signIn(email, password) 
  );

  useEffect(() => {
    if (!error) return;

    setShouldSubmit(false);
    alert('Invalid email and / or password.');
  }, [error]);

  return (
    <View style={styles.signInForm}>
      <TextInput
        label="Email"
        value={email}
        onChangeText={value => setEmail(value)}
      />
      <TextInput
        label="Password"
        value={password}
        onChangeText={value => setPassword(value)}
        secureTextEntry
      />

      <Button
        mode="contained"
        onPress={() => setShouldSubmit(true)}
      >
        Sign in
      </Button>
    </View>
  );
}
```

**Comment by dominictwlee** [comment]
The issue seems to be related to dispatch being reassigned during cleanup to `() => null`, which gets called before useEffect runs the 2nd time round.

Is there any reason why it's being reassigned that way, and can that be safely removed?

**Comment by shuding** [comment]
Should be fixed with #1396 🙏
